http-sink/metrics.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"fmt"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"net/http"
+)
+
+type MetricsPublisher struct {
+	payloadTotalCounter prometheus.Counter
+	payloadTotalSuccess prometheus.Counter
+	payloadTotalFailed  prometheus.Counter
+	payloadTotalDropped prometheus.Counter
+	payloadLatency      prometheus.Summary
+	payloadSize         prometheus.Summary
+	labels              map[string]string
+}
+
+func (mp *MetricsPublisher) registerMertics() {
+	mp.registerPayloadTotalCounter()
+	mp.registerPayloadTotalSuccess()
+	mp.registerPayloadTotalFailed()
+	mp.registerPayloadTotalDropped()
+	mp.registerPayloadSize()
+	mp.registerPayloadLatency()
+
+}
+
+func (mp *MetricsPublisher) IncreaseTotalCounter() {
+	mp.payloadTotalCounter.Inc()
+}
+
+func (mp *MetricsPublisher) IncreaseTotalSuccess() {
+	mp.payloadTotalSuccess.Inc()
+}
+func (mp *MetricsPublisher) IncreaseTotalFailed() {
+	mp.payloadTotalFailed.Inc()
+}
+func (mp *MetricsPublisher) IncreaseTotalDropped() {
+	mp.payloadTotalDropped.Inc()
+}
+
+func (mp *MetricsPublisher) UpdateSize(size float64) {
+	mp.payloadSize.Observe(size)
+}
+
+func (mp *MetricsPublisher) UpdateLatency(latency float64) {
+	mp.payloadLatency.Observe(latency)
+}
+
+func (mp *MetricsPublisher) registerPayloadTotalCounter() {
+	mp.payloadTotalCounter = promauto.NewCounter(prometheus.CounterOpts{
+		Name:        "payload_total_count",
+		Help:        "The total number of payload events",
+		ConstLabels: mp.labels,
+	})
+}
+
+func (mp *MetricsPublisher) registerPayloadTotalSuccess() {
+	mp.payloadTotalSuccess = promauto.NewCounter(prometheus.CounterOpts{
+		Name:        "payload_total_success",
```

Comment:
@vigith do you have a preference for a name?
total_payload_success
total_metrics_pushed

Reply:
payload is request? if yes, make it `total_requests_success`?